src/store/slices/web3/index.ts

```diff
@@ -23,14 +23,12 @@ const web3Slice = createSlice({
     setChain(state, action) {
```

Comment:
do you really need all these fine grained reducers?

wouldn't it make more sense to have reducers that combine more logic depending on the actual data changes?

e.g.: 
```
connectWallet(...) {
  // set connected logic
  // set chain logic
  // set address logic
}
```
of course this will only be valid if you get the data from one call.

It seems as the views have too much fine grained control over the state changes, which could lead to errors if other views "forget" or "misuse" those state changes

Reply:
It would be nice to have those yes, but the way wagmi works, we get events per speficic change. 
All of those reducers are called when we connect a wallet yes, but later you can only change the chain, or the wallet etc. We also make our app behave differently depending on what is changed in the wallet.

We could merge them and then instead of the new value, send an object with the change to the recuder like: `{account: '0xABC'}` and check what Object.keys we got in the reducer to later update the state, but then the reducer itself would have a lot of `if` in it and in the Redux history it would be harder to follow the changes.

Reply:
Ok I see, so we could only be able to create wrapper functions where applicable.
You can decide what makes sense here.